public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -257,7 +257,11 @@ export class ElasticDatasource {
         target.alias = this.templateSrv.replace(target.alias, options.scopedVars, 'lucene');
       }
 
-      const queryString = this.templateSrv.replace(target.query || '*', options.scopedVars, 'lucene');
+      let queryString = this.templateSrv.replace(target.query || '*', options.scopedVars, 'lucene');
```

Comment:
can you remove the `|| '*'` logic

Reply:
I initially thought that check was redundant too, but removing it can cause errors if target.query is undefined.  templateSrv.replace will return an undefined queryString if target.query is also undefined, which the queryBuilder won't like.  I only found this issue in one place, with a new installation, on the Home Dashboard, after completing the second step of 'Create your first data source', with an Elasticsearch Data Source, the built query will not have the required Elasticsearch "query" field.

However, a small modification to my PR accounts for both scenarios, and the || '*' logic can then be removed.  Will update the PR with the fix.

Excuse the long response, just wanted to document the resulting change.